tests/test_hyperthreading.py

```diff
@@ -0,0 +1,46 @@
+import json
+import logging
+import mock
+import pathlib
+import sys
+
+import mocked_xen_api_plugin
+
+sys.modules['XenAPIPlugin'] = mocked_xen_api_plugin
+sys.path.append(str(pathlib.Path(__file__).parent.resolve()) + '/../SOURCES/etc/xapi.d/plugins')
+
+from hyperthreading import get_hyperthreading
+@mock.patch('hyperthreading.run_command', autospec=True)
+class TestGetHyperthreading:
+    def test_hyperthreading(self, run_command):
+        run_command.side_effect = [{"stdout": "1"}, {"stdout": "2"}, {"stdout": "0"}, {"stdout": "-1"}]
+
+        res = get_hyperthreading(None, None)
+        assert not json.loads(res)
+
+        res = get_hyperthreading(None, None)
+        assert json.loads(res)
+
+        res = get_hyperthreading(None, None)
+        assert not json.loads(res)
+
+        res = get_hyperthreading(None, None)
+        assert not json.loads(res)
+
+        assert run_command.call_count == 4
+        run_command.assert_called_with(['xl', 'info', 'threads_per_core'])
+
+    def test_hyperthreading_error(self, run_command):
+        run_command.side_effect = [Exception('Error!')]
+
+        try:
+            get_hyperthreading(None, None)
+            logging.error("get_hyperthreading should have raised!")
+        except mocked_xen_api_plugin.Failure as e:
+            run_command.assert_called_with(['xl', 'info', 'threads_per_core'])
+            assert e.params[0] == '-1'
+            assert e.params[1] == 'Error!'
+            return
+        except Exception as e:
+            logging.error("Wrong exception raised: '{}', expected XenAPIPlugin.Failure".format(str(e)))
+        assert False
```

Comment:
You should probably use pytest.raises instead: https://docs.pytest.org/en/stable/assert.html#assertions-about-expected-exceptions